### DISCUSSION TITLE
rule is sometimes triggered by RfReceived and sometimes not

### DISCUSSION THREAD

**Discussion by BeryBurnout** [arendst/Tasmota#18523] [Q&A]
### PROBLEM DESCRIPTION
i need to convert some RF signals for my garage door, so i connected RF RX and TX, it seems to work, but the rule is sometimes not triggered by the received RF Data.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x ] Device used (e.g., Sonoff Basic): ___wemos D1 clone__
- [ ] Tasmota binary firmware version number used: _____
  - [x ] Pre-compiled
  - [ ] Self-compiled
- [ x] Flashing tools used: _web____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
15:08:27.841 CMD: Backlog Rule1
15:08:27.844 SRC: WebConsole from 192.168.178.34
15:08:27.846 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 5, Pld -99, Data 'Rule1'
15:08:27.871 SRC: Backlog
15:08:27.874 CMD: Grp 0, Cmd 'RULE', Idx 1, Len 0, Pld -99, Data ''
15:08:27.879 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":193,"Free":318,"Rules":"ON RFRECEIVED#DATA=0X1DDD5D5 DO RfSend {\"Data\":\"0x57E\",\"Bits\":12,\"Protocol\":8,\"Pulse\":718} ENDON ON RfReceived#Data=0x1DDD575 DO RfSend {\"Data\":\"0x57E\",\"Bits\":12,\"Protocol\":8,\"Pulse\":718} ENDON"}}


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:59:40.093 RUL: RFRECEIVED#DATA=0X1DDD575 performs "RfSend {"Data":"0x57E","Bits":12,"Protocol":8,"Pulse":718}"
14:59:40.095 SRC: Rule
14:59:40.097 CMD: Grp 0, Cmd 'RFSEND', Idx 1, Len 51, Pld -99, Data '{"Data":"0x57E","Bits":12,"Protocol":8,"Pulse":718}'
14:59:40.628 RSL: RESULT = {"RfSend":"Done"}
14:59:47.003 RFR: Data 0xDDD5D5 (14538197), Bits 24, Protocol 11, Delay 442
14:59:47.007 RSL: RESULT = {"Time":"2023-04-26T14:59:47","RfReceived":{"Data":"0xDDD5D5","Bits":24,"Protocol":11,"Pulse":442}}
14:59:47.103 RFR: Data 0x1DDD5D5 (31315413), Bits 25, Protocol 5, Delay 538
14:59:48.609 RFR: Data 0x1DDD5D5 (31315413), Bits 25, Protocol 5, Delay 540
14:59:48.911 WIF: Checking connection...
14:59:50.372 RFR: Data 0x1DDD5D5 (31315413), Bits 25, Protocol 5, Delay 537
14:59:50.375 RSL: RESULT = {"Time":"2023-04-26T14:59:50","RfReceived":{"Data":"0x1DDD5D5","Bits":25,"Protocol":5,"Pulse":537}}
14:59:50.393 RUL: RFRECEIVED#DATA=0X1DDD5D5 performs "RfSend {"Data":"0x57E","Bits":12,"Protocol":8,"Pulse":718}"
14:59:50.395 SRC: Rule
14:59:50.397 CMD: Grp 0, Cmd 'RFSEND', Idx 1, Len 51, Pld -99, Data '{"Data":"0x57E","Bits":12,"Protocol":8,"Pulse":718}'
14:59:50.928 RSL: RESULT = {"RfSend":"Done"}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
send randomly the RF-code, sometimes it works sometimes it doesn´t trigger the rule

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
everytime it receives the trigger code the rule should run

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
You have **two different** `RfReceived#Data` values, and your rule only matches one of them.
